nixos/modules/services/hardware/asusctl.nix

```
@@ -0,0 +1,482 @@
+        "AwakeLogo"
+        "ShutdownBar"
+        "SleepLogo"
+        "SleepKeyb"
+        "BootLogo"
+        "ShutdownKeyb"
+        "AwakeKeyb"
+        "BootKeyb"
+      ];
+    };
+  };
+
+  localAnimeConfig = {
+    system = [ ];
+    boot = [
+      {
+        ImageAnimation = {
+          file = "${cfg.package}/usr/share/asusd/anime/custom/sonic-run.gif";
+          scale = 0.9;
+          angle = 0.65;
+          translation = [
+            0.0
+            0.0
+          ];
+          time = {
+            Fade = {
+              fade_in = {
+                secs = 2;
+                nanos = 0;
+              };
+              show_for = {
+                secs = 2;
+                nanos = 0;
+              };
+              fade_out = {
+                secs = 2;
+                nanos = 0;
+              };
+            };
+          };
+          brightness = 1.0;
+        };
+      }
+    ];
+    wake = [
+      {
+        ImageAnimation = {
+          file = "${cfg.package}/usr/share/asusd/anime/custom/sonic-run.gif";
+          scale = 0.9;
+          angle = 0.65;
+          translation = [
+            0.0
+            0.0
+          ];
+          time = {
+            Fade = {
+              fade_in = {
+                secs = 2;
+                nanos = 0;
+              };
+              show_for = {
+                secs = 2;
+                nanos = 0;
+              };
+              fade_out = {
+                secs = 2;
+                nanos = 0;
+              };
+            };
+          };
+          brightness = 1.0;
+        };
+      }
+    ];
+    shutdown = [
+      {
+        ImageAnimation = {
+          file = "${cfg.package}/usr/share/asusd/anime/custom/sonic-wait.gif";
+          scale = 0.9;
+          angle = 0.0;
+          translation = [
+            3.0
+            2.0
+          ];
+          time = "Infinite";
+          brightness = 1.0;
+        };
+      }
+    ];
+    brightness = 1.0;
+    awake_enabled = true;
+    boot_anim_enabled = true;
+  };
+
+  mkAnimeConfig = generators.toJSON { } (recursiveUpdate localAnimeConfig cfg.animeConfig);
+  animeConfigFile = pkgs.writeText "anime.conf" (mkAnimeConfig);
+
+  mkAsusdConfig = generators.toJSON { } (recursiveUpdate localAsusdConfig cfg.asusdConfig);
+  asusdConfigFile = pkgs.writeText "asusd.conf" (mkAsusdConfig);
+
+  mkAuraConfig = generators.toJSON { } (recursiveUpdate localAuraConfig cfg.auraConfig);
+  auraConfigFile = pkgs.writeText "aura.conf" (mkAuraConfig);
+
+  profileConfigFile = pkgs.writeText "profile.conf" (cfg.profileConfig);
+  ledModesConfigFile = pkgs.writeText "asusd-ledmodes.conf" (cfg.ledModesConfig);
+in
+{
+  options = {
+
+    services.asusctl = {
+
+      enable = mkEnableOption (lib.mdDoc "Enable the asusctl service");
+
+      enableUserService = mkOption {
+        type = types.bool;
+        description = lib.mdDoc ''
+          Activate the asusd-user service.
+        '';
+        default = false;
```

```suggestion
        default = false;
        description = lib.mdDoc ''
          Activate the asusd-user service.
        '';
```
normally we put default before description

---

Done